text/0000-bionic-mixins.md

```diff
@@ -0,0 +1,75 @@
+This RFC proposes that we formalize the existing definition of `io.buildpacks.stacks.bionic` and define mixins on that stack to be Ubuntu 18.04 LTS package names (without version or architecture info) from official package mirrors. 
+
+In general, a stack is a contract that should be more like "Ubuntu Bionic" and less like "Platform XYZ's Re-distribution of Ubuntu Binoic."
+Stack images can still be provided by platforms/vendors, but they should use the standard CNB stack contracts where applicable.
+
+# Motivation
+[motivation]: #motivation
+
+This will allow:
+1. Buildpack authors to publish buildpacks that are compatible with more stack images without requiring those authors to explicitly specify vendor-specific stack IDs.
+2. Vendor-specific buildpacks to work together in the same build, as long as those buildpacks support `io.buildpacks.stacks.bionic`.
+3. App developers to create custom stack images that are compatible with any set of buildpacks that support `io.buildpacks.stacks.bionic`.
+
+# Example
+[example]: #example
+
+Say `org.cloudfoundry.stacks.cfbionicstack` is `ubuntu:bionic`, but with `build-essentials` in the build image and `git` in both build and run images.
+
+We would change the image metadata from:
+```
+run stackID: "org.cloudfoundry.stacks.cfbionicstack"
+run mixins: []
+build stackID: "org.cloudfoundry.stacks.cfbionicstack"
+build mixins: []
+```
+To:
+```
+run stackID: "io.buildpacks.stacks.bionic"
+run mixins: ["git"]
+build stackID: "io.buildpacks.stacks.bionic"
+build mixins: ["build:build-essentials", "git"]
+```
+
+And buildpack authors would update their `buildpack.toml` files from:
+```toml
+[[stacks]]
+id = "org.cloudfoundry.stacks.cfbionicstack"
+```
+To:
+```toml
+[[stacks]]
+id = "io.buildpacks.stacks.bionic"
+mixins = ["build:build-essentials", "git"] # NOTE: only need to list mixins they actually require!
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This could encourage buildpack authors and app developers to create a large number of unmaintained stack images that have different sets of Ubuntu packages.
+- Breaking change for buildpack authors once old stack images are removed.
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Neither the spec nor this RFC suggest whether or not stack authors or app developers are allowed to make additional changes to stack images without using mixin metadata.
+- This RFC doesn't propose a way for the project to describe changes to `ubuntu:bionic` that are not official LTS Ubuntu 18.04 packages.
+- When package groups (like `build-essential`) or dependent packages are installed, must stack images always specify all the packages added? Or is it only necessary when stack image authors want buildpacks to be able to depend on the dependent packages?
```

Comment:
The latter.  In order to make a mixin as broadly compatible as possible you'll probably want to list both the package group and all the packages that it includes.

Reply:
So your suggestion is that stack authors should add all packages to the mixin list, but are not required to? Would this imply that a given stack run/build image can have arbitrary contents that aren't specified by the ID or mixins, as long as those contents don't disappear when newer versions are published?